### PR TITLE
Add Wifi manager to main frontend

### PR DESCRIPTION
**Pull request by rafaellehmkuhl**
The wifi manager allows among other things:
- showing available networks (hidden or not) and it's properties;
- connecting to new networks (hidden or not);
- connect to existing networks (without pass);
- forgetting knows networks;
- forcing password change;
- disconnecting from current network;
- showing wifi adapter properties (including local IP, security type, signal level and others).

It is mobile-friendly, asynchronous and tries to follow interface patterns already familiar to most users.

Docker image available at `rafaellehmkuhl/companion-core:wifi-frontend`

Preview:
frontend

### CODE REVIEW

File: core/frontend/src/components/common/PasswordInput.vue [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  name: 'PasswordInput',
+  props: {
+    value: {
+      type: String,
+      default: '',
+    },
+  },
+  data () {
+    return {
+      show_password: '',
```
Comment (Williangalvani) [inline core/frontend/src/components/common/PasswordInput.vue:28]: why not a a boolean?

File: core/frontend/src/types/wifi.d.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    wifi_generation: int
+    pairwise_cipher: string
+    group_cipher: string
+    key_mgmt: string
+    wpa_state: string
+    ip_address: string
+    p2p_device_address: string
+    address: string
+    uuid: string
+}
```
Comment (rafaellehmkuhl) [inline core/frontend/src/types/wifi.d.ts:23]: number instead of int

File: core/frontend/src/types/wifi.d.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+export interface SavedNetwork {
+    networkid: number
+    ssid: string
+    bssid: string
+    flags: string
+}
+
+export class NetworkCredentials {
```
Comment (rafaellehmkuhl) [inline core/frontend/src/types/wifi.d.ts:40]: interface, not class

**Review by rafaellehmkuhl — COMMENTED** [review]
Just found some typing errors. Tracking.

File: core/frontend/src/components/wifi/WifiUpdater.vue [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+            index: ErrorIndex.WifiSavedFail,
+            message: `Could not fetch saved networks: ${error.message}.`,
+          })
+          wifi_store.setSavedNetworks([])
+        })
+    },
+    sleep (ms: number): Promise<void> {
+      return new Promise((resolve) => setTimeout(resolve, ms))
+    },
+    async updateNetworkStatus (): Promise<void> {
```
Comment (Williangalvani) [inline core/frontend/src/components/wifi/WifiUpdater.vue:112]: are these loops that run every 3s? There must be a better way to do this. 

wouldn't 
```
while (true) {
     await this.doStuff()
     await this.sleep(3000)
}
``` 
yield the same result, but in a more obvious way?

File: core/frontend/src/components/wifi/WifiUpdater.vue [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+      await this.fetchNetworkStatus()
+      await this.sleep(3000)
+      this.updateNetworkStatus()
+    },
+    async updateSavedNetworks (): Promise<void> {
```
Comment (Williangalvani) [inline core/frontend/src/components/wifi/WifiUpdater.vue:117]: I don't think we need to constantly fetch the saved and available networks.

**Review by Williangalvani — CHANGES_REQUESTED** [review]
This is looking good! I just did a broad review, I'll test it today.

File: core/frontend/src/types/wifi.d.ts (hunk 2 above)
Comment (patrickelectric) [reply core/frontend/src/types/wifi.d.ts:23]: Wait, `int` is not a valid type, how the tests are passing ??

File: core/frontend/src/store/common.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  name: 'common_store',
+})
+
+@Module
+export default class CommonStore extends VuexModule {
+  errors: Error[] = []
+  ignored_errors_indexes: ErrorIndex[] = []
+
+  @Mutation
+  openError (error: Error): void {
```
[the remark this answers is not in the document]
Comment (rafaellehmkuhl) [reply core/frontend/src/store/common.ts:17]: "add" seems to fit better, indeed, as "open" sounds like it will for sure open, which is not true if the user is ignoring this error.
Will change.

File: core/frontend/src/components/common/PasswordInput.vue (hunk 1 above)
Comment (rafaellehmkuhl) [reply core/frontend/src/components/common/PasswordInput.vue:28]: No idea why this is a string. It was a boolean for sure. More strange is the fact that this works. Fixing.

File: core/frontend/src/components/wifi/WifiUpdater.vue (hunk 4 above)
Comment (rafaellehmkuhl) [reply core/frontend/src/components/wifi/WifiUpdater.vue:112]: Better, indeed. Changing.

File: core/frontend/src/components/wifi/WifiUpdater.vue (hunk 5 above)
Comment (rafaellehmkuhl) [reply core/frontend/src/components/wifi/WifiUpdater.vue:117]: Didn't get the idea, exactly. You mean to just fetch at boot?

File: core/frontend/src/components/common/ErrorDialog.vue [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+const common_store: CommonStore = getModule(CommonStore)
+
+export default Vue.extend({
+  name: 'Error',
+  computed: {
+    error (): Error | null {
+      return common_store.current_error
+    },
+    error_message (): string {
```
Comment (patrickelectric) [inline core/frontend/src/components/common/ErrorDialog.vue:43]: `error` already provides the abstract for message, we should rely on the components that already exist. 
An abstraction just to access a member of a function that already return sounds as not a good interface practice.

File: core/frontend/src/components/common/ErrorDialog.vue [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+      return this.error === null ? '' : this.error.message
+    },
+    show_dialog: {
+      get (): boolean {
+        return this.error === null ? false : true
```
Comment (patrickelectric) [inline core/frontend/src/components/common/ErrorDialog.vue:48]:
```suggestion
        return this.error !== null
```

File: core/frontend/src/components/common/ErrorDialog.vue [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+      },
+      set (value: boolean): void {
```
Comment (patrickelectric) [inline core/frontend/src/components/common/ErrorDialog.vue:50]: `true` is not supported ? If so this should not receive arguments.

File: core/frontend/src/types/wifi.d.ts (hunk 2 above)
Comment (rafaellehmkuhl) [reply core/frontend/src/types/wifi.d.ts:23]: It's strange, indeed. Any idea?

File: core/frontend/src/components/common/PasswordInput.vue (hunk 1 above)
Comment (patrickelectric) [reply core/frontend/src/components/common/PasswordInput.vue:28]: How typescript failed to identify the wrong use of string here ??

File: core/frontend/src/components/common/SpinningLogo.vue [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+
+export default Vue.extend({
+  name: 'SpinningLogo',
+})
+</script>
+
+<style>
+  .lds-dual-ring {
+    display: block;
+    width: 80px;
```
Comment (patrickelectric) [inline core/frontend/src/components/common/SpinningLogo.vue:19]: avoid `px`, use `em` to support high definition displays.

File: core/frontend/src/components/wifi/WifiUpdater.vue (hunk 5 above)
Comment (Williangalvani) [reply core/frontend/src/components/wifi/WifiUpdater.vue:117]: I still haven't run it here, but I got the impression that this is always running.
we should have it scanning only when the wifi widget is open.

File: core/frontend/src/components/wifi/ConnectionDialog.vue [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+      return this.is_hidden ? this.inputed_ssid : this.network.ssid
+    },
+    is_hidden (): boolean {
+      return this.network.ssid === null || this.network.ssid === ''
+    },
+    show_password_input_box (): boolean {
+      if (this.force_password) {
+        return true
+      }
+      else if (this.network.saved || !this.network.locked) {
```
Comment (patrickelectric) [inline core/frontend/src/components/wifi/ConnectionDialog.vue:124]: remove else after return

File: core/frontend/src/components/wifi/ConnectionDialog.vue [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+      this.password = ''
+      this.emitChange(false)
+
+      const connect_response = await axios({
+        method: 'post',
+        url: `${wifi_store.API_URL}/connect`,
+        timeout: 500,
+        data: {ssid: this.real_ssid, password},
+        params: { hidden: this.is_hidden },
+      })
```
Comment (patrickelectric) [inline core/frontend/src/components/wifi/ConnectionDialog.vue:142]: why not use `then/catch`  for better readability?

File: core/frontend/src/types/wifi.d.ts (hunk 2 above)
Comment (Williangalvani) [reply core/frontend/src/types/wifi.d.ts:23]: we should add a test to test the tests :eyes:

File: core/frontend/src/components/wifi/ConnectionDialog.vue [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+        .then(() => {
+          common_store.closeError(ErrorIndex.WifiForgetFail)
+        })
+        .catch((error) => {
+          common_store.openError({
+            index: ErrorIndex.WifiForgetFail,
+            message: `Could not remove saved wifi network: ${error.message}.`})
+        })
+    },
+    emitChange (state: boolean) {
```
Comment (patrickelectric) [inline core/frontend/src/components/wifi/ConnectionDialog.vue:170]: What changed ? The function should have a better name

File: core/frontend/src/components/wifi/ConnectionDialog.vue [hunk 14]
```diff
@@ -0,0 +1,178 @@
...
+        params: { hidden: this.is_hidden },
+      })
+
+      if (connect_response.status != 200) {
+        common_store.openError({index: ErrorIndex.WifiConnectFail, message: 'Could not connect to wifi network.'})
+        return
+      }
+      common_store.closeError(ErrorIndex.WifiConnectFail)
+
+      wifi_store.setNetworkStatus(null)
```
Comment (patrickelectric) [inline core/frontend/src/components/wifi/ConnectionDialog.vue:150]: What is a null state ? This should be more clear.

File: core/frontend/src/components/wifi/NetworkCard.vue [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+      // | -70 dBm         | Okay      | Minimum signal strength for reliable packet delivery.                                                                                      |
+      // | -80 dBm         | Not Good  | Minimum signal strength for basic connectivity. Packet delivery may be unreliable.                                                         |
+      // | -90 dBm         | Unusable  | Approaching or drowning in the noise floor. Any functionality is highly unlikely.                                                           |
+      // Reference: metageek.com/training/resources/wifi-signal-strength-basics.html
+      /* eslint-enable */
+
+      if (this.network.signal >= -67) return 'mdi-wifi-strength-4'
+      if (this.network.signal >= -70) return 'mdi-wifi-strength-3'
+      if (this.network.signal >= -80) return 'mdi-wifi-strength-2'
+      return 'mdi-wifi-strength-1'
```
Comment (patrickelectric) [inline core/frontend/src/components/wifi/NetworkCard.vue:74]: since this is "unusable", we should use `mdi-wifi-strength-outline`, the others we could do -1 and also include >= 30 with strength-4

File: core/frontend/src/components/wifi/WifiManager.vue [hunk 16]
```diff
@@ -0,0 +1,122 @@
...
+  data () {
+    return {
+      selected_network: null as Network | null,
+      show_connection_dialog: false,
+      show_disconnection_dialog: false,
+    }
+  },
+  computed: {
+    are_networks_available (): boolean {
+      return wifi_store.available_networks.length === 0 ? false : true
```
Comment (patrickelectric) [inline core/frontend/src/components/wifi/WifiManager.vue:76]: As general tip, if you are checking for true/false, you don't need to return true or false, this is a bad practice.

File: core/frontend/src/components/wifi/WifiUpdater.vue [hunk 17]
```diff
@@ -0,0 +1,129 @@
...
+          )
+          const saved_network = wifi_store.saved_networks.find(
+            (network) => network.ssid === response.data.ssid,
+          )
+
+          wifi_store.setCurrentNetwork({
+            ssid: response.data.ssid,
+            signal: scanned_network ? scanned_network.signal : 0,
+            locked: response.data.key_mgmt.includes('WPA'),
+            saved: saved_network ? true : false,
```
Comment (patrickelectric) [inline core/frontend/src/components/wifi/WifiUpdater.vue:52]: As said.

File: core/frontend/src/components/wifi/WifiUpdater.vue [hunk 18]
```diff
@@ -0,0 +1,129 @@
...
+              locked: network.flags.includes('WPA'),
+              saved: saved_networks_ssids.includes(network.ssid),
+            }
+          })
+          wifi_store.setAvailableNetworks(available_networks)
+        })
+        .catch((error) => {
+          common_store.openError({
+            index: ErrorIndex.WifiScanFail,
+            message: `Could not scan wifi networks: ${error.message}`,
```
Comment (patrickelectric) [inline core/frontend/src/components/wifi/WifiUpdater.vue:86]: `Could not scan for Wi-Fi networks`

File: core/frontend/src/types/common.ts [hunk 19]
```diff
@@ -0,0 +1,13 @@
+export interface Error {
+    index: ErrorIndex
+    message: string
+}
+
+export enum ErrorIndex {
+    WifiForgetFail = 'wifi_forget_fail',
```
Comment (patrickelectric) [inline core/frontend/src/types/common.ts:7]: Why the strings are necessary for this enum ?

File: core/frontend/src/types/common.ts [hunk 20]
```diff
@@ -0,0 +1,13 @@
+export interface Error {
```
Comment (patrickelectric) [inline core/frontend/src/types/common.ts:1]: This should be in an Error file and not in a common.ts

File: core/frontend/src/store/common.ts [hunk 21]
```diff
@@ -0,0 +1,44 @@
...
+    if (this.ignored_errors_indexes.includes(error.index)) {
+      console.log(`Errors of type ${error.index} are being ignored. Not opening error dialog.`)
```
Comment (patrickelectric) [inline core/frontend/src/store/common.ts:19]: console.warning ?

File: core/frontend/src/store/common.ts [hunk 22]
```diff
@@ -0,0 +1,44 @@
...
+      return
+    }
+    if (this.errors.includes(error)) {
+      console.log(`Error of type ${error.index} is already open.`)
```
Comment (patrickelectric) [inline core/frontend/src/store/common.ts:23]: console.error ?

**Review by patrickelectric — CHANGES_REQUESTED** [review]

File: core/frontend/src/types/wifi.d.ts (hunk 2 above)
Comment (patrickelectric) [reply core/frontend/src/types/wifi.d.ts:23]: This a must thing to fix and do it asap, if the typescript checks are not working, something is really wrong with the project.

File: core/frontend/src/components/wifi/WifiUpdater.vue (hunk 5 above)
Comment (rafaellehmkuhl) [reply core/frontend/src/components/wifi/WifiUpdater.vue:117]: It's an option, for sure.
Was like that before, in fact, but I changed.

File: core/frontend/src/components/common/ErrorDialog.vue (hunk 7 above)
Comment (rafaellehmkuhl) [reply core/frontend/src/components/common/ErrorDialog.vue:43]: Will check on using it.

File: core/frontend/src/components/common/ErrorDialog.vue (hunk 9 above)
Comment (rafaellehmkuhl) [reply core/frontend/src/components/common/ErrorDialog.vue:50]: Didn't get it. get/set for are an existing interface that I'm implementing.

**Comment by patrickelectric** [comment]
I'm receiving error messages every time that I open the widget 
image
image

**Comment by patrickelectric** [comment]
Wifi appears as saved but asks for password again
image

**Comment by patrickelectric** [comment]
The wifi popup appears with a unnecessary scrollbar and unnecessary nerd information, we could add it as an "i" information icon or dropdown.

image

**Comment by rafaellehmkuhl** [comment]
> I'm receiving error messages every time that I open the widget
> image
> image
> image

Could you send a print of the network tab on the inspector?
Are you receiving this constantly? From the start or after some time?

**Comment by rafaellehmkuhl** [comment]
> Wifi appears as saved but asks for password again
> image

You're sure you didn't click the "force new password" button?
Could you post here the JSON that comes from both 'scan' and 'saved'?

**Comment by rafaellehmkuhl** [comment]
> The wifi popup appears with a unnecessary scrollbar and unnecessary nerd information, we could add it as an "i" information icon or dropdown.
> 
> image

Sounds like a good idea.

**Comment by patrickelectric** [comment]
> > I'm receiving error messages every time that I open the widget
> > image
> > image
> > image
> 
> Could you send a print of the network tab on the inspector?
> Are you receiving this constantly? From the start or after some time?

When I click in connect:
image
When I click to connect in another network:
image

**Comment by patrickelectric** [comment]
> > Wifi appears as saved but asks for password again
> > image
> 
> You're sure you didn't click the "force new password" button?
> Could you post here the JSON that comes from both 'scan' and 'saved'?

saved:
image


scan:
image

**Comment by rafaellehmkuhl** [comment]
> > > I'm receiving error messages every time that I open the widget
> > > image
> > > image
> > > image
> > 
> > 
> > Could you send a print of the network tab on the inspector?
> > Are you receiving this constantly? From the start or after some time?
> 
> When I click in connect:
> image
> When I click to connect in another network:
> image

Cool! That really helps!
It could be that when connecting the WPA Supplicant gets busier and delivers the other results with more delay than I was expecting.
I will try to reproduce it here to see if that's just a matter of changing the timeouts or if we could change the logic to improve that.

**Comment by rafaellehmkuhl** [comment]
> > > Wifi appears as saved but asks for password again
> > > image
> > 
> > 
> > You're sure you didn't click the "force new password" button?
> > Could you post here the JSON that comes from both 'scan' and 'saved'?
> 
> saved:
> image
> 
> scan:
> image

This one I'm not understanding why it happens. 
Does it stays like this or it's a momentary problem?
Do the other saved networks have the same behavior? I tried with some of them here and couldn't reproduce.

File: core/frontend/src/types/wifi.d.ts (hunk 2 above)
Comment (rafaellehmkuhl) [reply core/frontend/src/types/wifi.d.ts:23]: Apparently what's happening is that the 'int' was casted to 'unknown', thus accepting literally anything.
[There's a rule](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-explicit-any.md) for setting that.I will take a better look to see if it's the responsible for this behavior.

File: core/frontend/src/components/common/PasswordInput.vue (hunk 1 above)
Comment (rafaellehmkuhl) [reply core/frontend/src/components/common/PasswordInput.vue:28]: What happened what that, when using `show_password ?`, the empty string was being cast to a boolean, returning `false`, and on the first call to `show_password = !show_password`, it was turned into a *de facto* boolean. All the logic works fine and the operations are valid.

The problem I see is that the Typescript linter was unable to see that a variable was having its type altered on runtime. This happened because this toggle operation happened inside the `<template>`. When I move the operation to a method, inside the `<script>` tag, the linter is able to catch the type-change mistake and forces me to declare it initially as a boolean.

This enforces that we should avoid any operation inside the template, and use `computed` and `methods` whenever possible.

File: core/frontend/src/components/common/SpinningLogo.vue (hunk 10 above)
Comment (rafaellehmkuhl) [reply core/frontend/src/components/common/SpinningLogo.vue:19]: Using `%` instead, as it's the recommended way to scale relative to parent elements. `em` are font-size dependent, so it wouldn't be the best choice here.

File: core/frontend/src/components/wifi/ConnectionDialog.vue (hunk 12 above)
Comment (rafaellehmkuhl) [reply core/frontend/src/components/wifi/ConnectionDialog.vue:142]: Forgot to change this one.

File: core/frontend/src/components/common/SpinningLogo.vue (hunk 10 above)
Comment (patrickelectric) [reply core/frontend/src/components/common/SpinningLogo.vue:19]: `em` are not font-size dependent, they are relative to resolution, as fonts.

> Pixel is a static measurement, while percent and EM are relative measurements. The size of an EM or percent depends on its parent. If the text size of body is 16 pixels, then 150% or 1.5 EM will be 24 pixels (1.5 * 16).

Comment (rafaellehmkuhl) [reply core/frontend/src/components/common/SpinningLogo.vue:19]: https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Values_and_units

Relative length units section.

File: core/frontend/src/types/common.ts (hunk 19 above)
Comment (rafaellehmkuhl) [reply core/frontend/src/types/common.ts:7]: They are not. Removing.

File: core/frontend/src/components/wifi/ConnectionDialog.vue (hunk 14 above)
Comment (rafaellehmkuhl) [reply core/frontend/src/components/wifi/ConnectionDialog.vue:150]: It's a `null` only when there's no status information available, as in the case of API request errors.

File: core/frontend/src/components/wifi/NetworkCard.vue (hunk 15 above)
Comment (rafaellehmkuhl) [reply core/frontend/src/components/wifi/NetworkCard.vue:74]: Liked the idea for the outline, but got confused with the other one. Are you suggesting this?

```javascript
if (this.network.signal >= -30) return 'mdi-wifi-strength-4'
if (this.network.signal >= -67) return 'mdi-wifi-strength-3'
if (this.network.signal >= -70) return 'mdi-wifi-strength-2'
if (this.network.signal >= -80) return 'mdi-wifi-strength-1'
return 'mdi-wifi-strength-outline'
```

File: core/frontend/src/components/wifi/WifiManager.vue (hunk 16 above)
Comment (rafaellehmkuhl) [reply core/frontend/src/components/wifi/WifiManager.vue:76]: True. I usually write the whole phrase as a sanity-check and fix it after, but sometimes I forgot to fix rs

File: core/frontend/src/components/wifi/WifiUpdater.vue (hunk 4 above)
Comment (rafaellehmkuhl) [reply core/frontend/src/components/wifi/WifiUpdater.vue:112]: Will, in fact this approach will lock javascript in the while block. Do you know some better alternative considering the async/await approach?

**Comment by rafaellehmkuhl** [comment]
@patrickelectric can you test if you're still experiencing issues regarding the timeouts with the new image?

File: core/frontend/src/components/wifi/WifiUpdater.vue [hunk 23]
```diff
@@ -0,0 +1,120 @@
...
+    },
+    async startNetworkStatusUpdateLoop (): Promise<void> {
+      await this.fetchNetworkStatus()
+      await this.sleep(5000)
+      this.startNetworkStatusUpdateLoop()
+    },
+    async startNetworksScanLoop (): Promise<void> {
+      await this.fetchAvailableNetworks()
+      await this.sleep(5000)
+      this.startNetworksScanLoop()
```
Comment (patrickelectric) [inline core/frontend/src/components/wifi/WifiUpdater.vue:116]: We could abstract this in a better way.  maybe a periodicRequest function ?

**Review by patrickelectric — CHANGES_REQUESTED** [review]
there is still some problems related to the behavior, we did talk via chat.

File: core/frontend/src/components/wifi/WifiUpdater.vue (hunk 23 above)
Comment (rafaellehmkuhl) [reply core/frontend/src/components/wifi/WifiUpdater.vue:116]: Good idea!

File: core/frontend/src/components/wifi/DisconnectionDialog.vue [hunk 24]
```diff
@@ -0,0 +1,127 @@
...
+    show: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  computed: {
+    show_more_info_icon (): string {
+      return this.show_more_info ? 'mdi-chevron-up' : 'mdi-chevron-down'
+    },
+    connection_info (): Network | Network & WifiStatus {
```
[the remark this answers is not in the document]
Comment (rafaellehmkuhl) [reply core/frontend/src/components/wifi/DisconnectionDialog.vue]: Although I understand the suggestion, the problem is that some properties like the SSID are duplicated on the two. That's why I merged then in this way. 

The type is very strict check as it is. There's only the possibility of returning a Network object alone or a merged Network+WifiStatus object. If I add or remove any property, Typescript blames me.

For clarity of syntax I've added brackets around the `Network & WifiStatus`.

File: core/frontend/src/components/wifi/ConnectionDialog.vue [hunk 25]
```diff
@@ -0,0 +1,199 @@
...
+      const password = this.password
+      this.password = ''
+      this.showDialog(false)
+
+      await axios({
+        method: 'post',
+        url: `${wifi_store.API_URL}/connect`,
+        timeout: 2000,
+        data: {ssid: this.real_ssid, password},
+        params: { hidden: this.is_hidden },
```
[the remark this answers is not in the document]
Comment (patrickelectric) [reply core/frontend/src/components/wifi/ConnectionDialog.vue:177]: Just to be sure, this was fixed manually or via eslint ?

[the remark this answers is not in the document]
Comment (rafaellehmkuhl) [reply core/frontend/src/components/wifi/ConnectionDialog.vue:177]: Let me check.

[the remark this answers is not in the document]
Comment (patrickelectric) [reply core/frontend/src/components/wifi/ConnectionDialog.vue:177]: Let me rephrase that, was eslint able to detect the style problem ?

[the remark this answers is not in the document]
Comment (rafaellehmkuhl) [reply core/frontend/src/components/wifi/ConnectionDialog.vue:177]: If you're talking about the spaces around the curly braces, this is being done automatically. Just tried to make it bad and eslint doesn't allow.

**Comment by rafaellehmkuhl** [comment]
I'm building a new image for you guys to test. Should be ready in 10 minutes.

File: core/frontend/src/components/wifi/ConnectionDialog.vue (hunk 25 above)
[the remark this answers is not in the document]
Comment (rafaellehmkuhl) [reply core/frontend/src/components/wifi/ConnectionDialog.vue:177]: > Let me rephrase that, was eslint able to detect the style problem ?

Yes.

**Comment by rafaellehmkuhl** [comment]
> I'm building a new image for you guys to test. Should be ready in 10 minutes.

Already available at `rafaellehmkuhl/companion-core:wifi-frontend`.

**Review by patrickelectric — APPROVED** [review]
